lib/apartmentex/tenant_actions.ex

```diff
@@ -21,6 +33,33 @@ defmodule Apartmentex.TenantActions do
     end
   end
 
+  defp migrate_and_return_status(repo, tenant, direction, opts) do
+    prefix = build_prefix(tenant)
+
+    {status, versions} = handle_database_exceptions fn ->
+      opts_with_prefix = Keyword.put(opts, :prefix, prefix)
+      Ecto.Migrator.run(
+        repo,
+        tenant_migration_folder(repo),
+        direction,
+        opts_with_prefix
+      )
+    end
+
+    {status, prefix, versions}
+  end
+
+  defp handle_database_exceptions(fun) do
+    try do
+      {:ok, fun.()}
+    rescue
+      e in Postgrex.Error ->
+        {:error, Postgrex.Error.message(e)}
+      e in Mariaex.Error ->
+        {:error, Mariaex.Error.message(e)}
```

Comment:
You are right about Mysql- database drop and create cause commit and can't be run in a transaction. I think the amount of people using Phoenix and MySql is small and shrinking anyway- mostly I think its people applying it to legacy databases which wouldn't have much use for apartmentex anyway.  I think we're ok just testing Postgres, but I'll make an example app with Mysql just to see if it seems to work.